builtin/providers/aws/resource_aws_instance.go

```diff
@@ -475,6 +475,9 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("subnet_id", instance.SubnetId)
 	}
 	d.Set("ebs_optimized", instance.EbsOptimized)
+	if instance.SubnetId != nil && *instance.SubnetId != "" {
+		d.Set("source_dest_check", instance.SourceDestCheck)
```

Comment:
Should we forcefully set it to ``true`` if there isn't a subnet, rather than just letting it stick on whatever it's already set to? Doesn't really matter I guess, since the ``Update`` path won't do anything with it, but it could avoid leaving a misleading value in the state.

Alternatively the path in ``Update`` could return an error if ``source_dest_check`` is set to ``false`` when subnet isn't set, to make it more obvious that this combination is invalid.